test/e2e/volume_provisioning.go

```diff
@@ -54,12 +54,16 @@ func testDynamicProvisioning(client clientset.Interface, claim *v1.PersistentVol
 
 	// Check sizes
 	expectedCapacity := resource.MustParse(expectedSize)
+	if framework.ProviderIs("vsphere") {
+		// vsphere provider does not allocate volumes in 1GiB chunks
+		expectedCapacity = resource.MustParse(requestedSize)
```

Comment:
Can you also change the comment on L37-38 to be more generic and recognize exceptions? Would be be better to have expectedSize `const` per provider?